tools/copyblocks/main.go

```diff
@@ -171,13 +187,72 @@ func copyBlocks(ctx context.Context, cfg config, logger log.Logger, m *metrics)
 		disabledUsers[u] = struct{}{}
 	}
 
-	client, err := storage.NewClient(ctx)
-	if err != nil {
-		return errors.Wrapf(err, "failed to create client")
-	}
+	var sourceBucket, destBucket bucket
+	switch strings.ToLower(cfg.service) {
```

Comment:
This method (`copyBlocks()`) is getting quite long - could we break it into smaller pieces? For example, this initialisation of the source and destination buckets could be extracted to its own method.

Reply:
Very fair point, I refactored to make this cleaner.